include/zephyr/kernel.h

```diff
@@ -4574,6 +4575,30 @@ int k_msgq_cleanup(struct k_msgq *msgq);
  */
 __syscall int k_msgq_put(struct k_msgq *msgq, const void *data, k_timeout_t timeout);
 
+/**
+ * @brief Send a message to front of a message queue.
+ *
+ * This routine sends a message to message queue @a q, inserting it before any
+ * other messages already in the queue.
+ *
+ * @note The message content is copied from @a data into @a msgq and the @a data
+ * pointer is not retained, so the message content will not be modified
+ * by this function.
+ *
+ * @funcprops \isr_ok
+ *
+ * @param msgq Address of the message queue.
+ * @param data Pointer to the message.
+ * @param timeout Non-negative waiting period to add the message,
```

Comment:
I realize that this the timeout description was copy-pasted, but I think that description is out of date. Many of the other kernel APIs that use a timeout no longer use "Non-negative" as that is used to represent an absolute timeout.

I recommend dropping "Non-negative".